archive/map-reduce.md

```diff
@@ -0,0 +1,116 @@
+# Map-reduce
+
+## Motivation
+
+There are hundreds of programming models to handle some data sets. But the Google has some problems to handle their data sets.
+
+- The input data is large.
+- The computations have to be distributed across more than hundreds of machines.
+
+To handle those data sets, the below detalis must be implemented.
+
+- how to parallelized the computation
+- how to distribute the data
+- how to handle failures
+
+After Google implemented hundreds of such programming models, they made an abstraction of them, that is Map-reduce model.
+
+## Model definition
+
+The programming model can be defined as the input and output of its execution. There are three execution model on Map-reduce model, that are the full program, map function and reduce function.
+
+- Full execution
+  - Input: a set of key/value pair
+  - Output: a set of key/value pair
+- Map: the function of producing intermediate key/value pair
+  - Input: a key/value pair
```

Comment:
```suggestion
  - Input: a set of key/value pair
```

Reply:
From the paper, the map will accept an input pair.

![image](https://github.com/user-attachments/assets/d294dc98-909f-4453-839d-1554c1f345af)

Reply:
Oh, I see. Thank you. Then, why ... the program - Full execution - requires set?
```
- Full execution
  - Input: a set of key/value pair
  - Output: a set of key/value pair
```
https://github.com/TheWeaVer/lia/pull/26/files#diff-d6e6fe5544ed282cd2bd3e96db96991677b08a950ba2ca1ecc6f93553d8b1b9dR22-R24

Reply:
The full execution is a computation. The right above lines of the above image. Let me change the `full execution` to `computation`. Sorry to make a confusion. 😢

Reply:
Ah, then I understand that the full procedure ! Thank you !

Reply:
This is applied at a3852f5afa4573109179c9debebc43562dcf1c97.